File: argentum-ui/src/app/common/rest-service/test-data/transactions.ts
import {Transaction} from '../../model/transaction';
import {Guests} from './guests';
import {Orders} from './orders';

export namespace Transactions {
  export const TX1 = new Transaction(
    20010,
    new Date('2019-12-31T22:05:00Z'),
    Guests.ROBY,
    undefined,
    9.00,
    false,
    'default',
    null,
    false
  );

  export const TX_COAT_CHECK_1 = new Transaction(
    20020,
    new Date('2019-12-31T22:07:00Z'),
    Guests.ROBY,
    undefined,
    -1.00,
    false,
    'order',
    Orders.TAG_REGISTRATION_TWO,
    false
  );

  export const TX2 = new Transaction(
    20030,
    new Date('2019-12-31T22:07:30Z'),
    Guests.SHEELAH,
    undefined,
    5.00,
    true,
    'default',
    null,
    false
  );

  export const TX_COAT_CHECK_2 = new Transaction(
    20040,
    new Date('2019-12-31T22:09:00Z'),
    Guests.SHEELAH,
    undefined,
    -1.00,
    false,
    'order',
    Orders.TAG_REGISTRATION_THREE,
    false
  );

  export const TX_ORDER_1 = new Transaction(
    20050,
    new Date('2019-12-31T22:10:00Z'),
    Guests.ROBY,
    undefined,
    -3.00,
    false,
    'order',
    Orders.ONE_WATER_PLUS_TIP,
    false
  );

  export const TX3 = new Transaction(
    20060,
    new Date('2019-12-31T22:30:00Z'),
    Guests.SHEELAH,
    undefined,
    -5.00,
    false,
    'default',
    null,
    true
  );

  export const ALL = [TX1, TX_COAT_CHECK_1, TX2, TX_COAT_CHECK_2, TX_ORDER_1, TX3];

  export const TX1_BY_CARD = new Transaction(
    20010,
    new Date('2019-12-31T22:05:00Z'),
    undefined,
    undefined,
    9.00,
    false,
    'default',
    null,
    false
  );

  export const TX_COAT_CHECK_1_BY_CARD = new Transaction(
    20020,
    new Date('2019-12-31T22:07:00Z'),
    undefined,
    undefined,
    -1.00,
    false,
    'order',
    Orders.TAG_REGISTRATION_TWO,
    false
  );

  export const TX_ORDER_1_BY_CARD = new Transaction(
    20050,
    new Date('2019-12-31T22:10:00Z'),
    undefined,
    undefined,
    -3.00,
    false,
    'order',
    Orders.ONE_WATER_PLUS_TIP,
    false
  );

  export const TX4_REFERENCE = new Transaction(
    20070,
    new Date('2019-12-31T22:35:30Z'),
    undefined,
    undefined,
    5.00,
    false,
    'default',
    null,
    true
  );

  export const TX3_PATCHED_REFERENCE = new Transaction(
    20061,
    new Date('2019-12-31T22:30:05Z'),
    undefined,
    undefined,
    -5.00,
    false,
    'default',
    null,
    false
  );
}

File: argentum-api/api/tests/data/transactions.py
from decimal import Decimal

from django.utils.dateparse import parse_datetime

from api.models.transaction import Transaction
from api.tests.data.guests import TestGuests
from api.tests.data.orders import TestOrders
from api.tests.utils.test_objects import TestObjects


class TestTransactions(TestObjects):
    MODEL = Transaction

    TX1: MODEL
    TX_COAT_CHECK_1: MODEL
    TX2: MODEL
    TX_COAT_CHECK_2: MODEL
    TX_ORDER_1: MODEL
    TX3: MODEL

    TX3_COMMITTED: MODEL
    TX4: MODEL
    TX_ORDER_2: MODEL
    TX_CANCEL_1: MODEL
    TX_CANCEL_2: MODEL
    TX_COAT_CHECK_3: MODEL

    @classmethod
    def init(cls):
        cls.TX1 = cls.MODEL(
            id=20010,
            time=parse_datetime('2019-12-31T22:05:00Z'),
            guest=TestGuests.ROBY,
            value=Decimal('9.00'),
            ignore_bonus=False,
            description='default',
            order=None,
            pending=False
        )

        cls.TX_COAT_CHECK_1 = cls.MODEL(
            id=20020,
            time=parse_datetime('2019-12-31T22:07:00Z'),
            guest=TestGuests.ROBY,
            value=Decimal('-1.00'),
            ignore_bonus=False,
            description='order',
            order=TestOrders.TAG_REGISTRATION_TWO,
            pending=False
        )

        cls.TX2 = cls.MODEL(
            id=20030,
            time=parse_datetime('2019-12-31T22:07:30Z'),
            guest=TestGuests.SHEELAH,
            value=Decimal('5.00'),
            ignore_bonus=True,
            description='default',
            order=None,
            pending=False
        )

        cls.TX_COAT_CHECK_2 = cls.MODEL(
            id=20040,
            time=parse_datetime('2019-12-31T22:09:00Z'),
            guest=TestGuests.SHEELAH,
            value=Decimal('-1.00'),
            ignore_bonus=False,
            description='order',
            order=TestOrders.TAG_REGISTRATION_THREE,
            pending=False
        )

        cls.TX_ORDER_1 = cls.MODEL(
            id=20050,
            time=parse_datetime('2019-12-31T22:10:00Z'),
            guest=TestGuests.ROBY,
            value=Decimal('-3.00'),
            ignore_bonus=False,
            description='order',
            order=TestOrders.ONE_WATER_PLUS_TIP,
            pending=False
        )

        cls.TX3 = cls.MODEL(
            id=20060,
            time=parse_datetime('2019-12-31T22:30:00Z'),
            guest=TestGuests.SHEELAH,
            value=Decimal('-5.00'),
            ignore_bonus=False,
            description='default',
            order=None,
            pending=True
        )

        cls.SAVED = [cls.TX1, cls.TX_COAT_CHECK_1, cls.TX2, cls.TX_COAT_CHECK_2, cls.TX_ORDER_1, cls.TX3]

        cls.TX3_COMMITTED = cls.MODEL(
            id=20061,
            guest=TestGuests.SHEELAH,
            value=Decimal('-5.00'),
            ignore_bonus=False,
            description='default',
            order=None,
            pending=False
        )

        cls.TX4 = cls.MODEL(
            id=20070,
            guest=TestGuests.ROBY,
            value=Decimal('5.00'),
            ignore_bonus=False,
            description='default',
            order=None,
            pending=True
        )

        cls.TX_ORDER_2 = cls.MODEL(
            id=20080,
            guest=TestGuests.SHEELAH,
            value=Decimal('-7.00'),
            ignore_bonus=False,
            description='order',
            order=TestOrders.TWO_COKES_PLUS_TIP,
            pending=False
        )

        cls.TX_CANCEL_1 = cls.MODEL(
            id=20090,
            guest=TestGuests.ROBY,
            value=Decimal('0.15'),
            ignore_bonus=False,
            description='cancel',
            order=TestOrders.ONE_WATER_PLUS_TIP,
            pending=False
        )

        cls.TX_CANCEL_2 = cls.MODEL(
            id=20100,
            guest=TestGuests.ROBY,
            value=Decimal('2.40'),
            ignore_bonus=False,
            description='cancel',
            order=TestOrders.ONE_WATER_PLUS_TIP,
            pending=False
        )

        cls.TX_COAT_CHECK_3 = cls.MODEL(
            id=20110,
            guest=TestGuests.ROBY,
            value=Decimal('-1.00'),
            ignore_bonus=False,
            description='order',
            order=TestOrders.TAG_REGISTRATION_FOUR,
            pending=False
        )

        cls.UNSAVED = [
            cls.TX3_COMMITTED,
            cls.TX4,
            cls.TX_ORDER_2,
            cls.TX_CANCEL_1,
            cls.TX_CANCEL_2,
            cls.TX_COAT_CHECK_3
        ]
